### PR TITLE
Add "create group invitation" http-service

### PR DESCRIPTION
**Pull request by GabrielVidal1**
Added the http-service for the group-invite-users component.
This service is calling the [groupInvitationsCreate](https://france-ioi.github.io/algorea-devdoc/api/#operation/groupInvitationsCreate) in the Backend.

### CODE REVIEW

**Review by smadbe — CHANGES_REQUESTED** [review]
Very good.  👍  
I would suggest a change to make you component cleaner (which was not done in `RequestActionsService` but probably should as well):
It is not really great to obtain a `Map<string,any>` for the complement as any could be anything. After your two rx "map(..)"'s, you should again your `Map<string, any>` to `Map<string, InvitationResult>` where `InvitationResult` would be a union type of what is supposed in the service (the enum  [cycle, invalid, success, unchanged, not_found]) but reinterpreted for our needs, so for instance mapping to 'success'|'error'|'already_invited'|'not_found', and throwing an error if you get something unexpected.

File: src/app/modules/group/http-services/create-group-invitations.service.ts [hunk 1]
```diff
@@ -2,9 +2,15 @@ import { Injectable } from '@angular/core';
...
-import { ActionResponse, successData, objectToMap } from 'src/app/shared/http-services/action-response';
+import { ActionResponse, successData } from 'src/app/shared/http-services/action-response';
 import { map } from 'rxjs/operators';
 
+export enum InvitationResult {
+  success,
+  error,
+  already_invited,
+  not_found,
+}
```
Comment (smadbe) [inline src/app/modules/group/http-services/create-group-invitations.service.ts:13]: I was suggesting an union type but in this case probably an enum is better, right 👍 
However, enum options should be in PascalCase, so `Success`,  `AlreadyInvited`, ...

File: src/app/modules/group/http-services/create-group-invitations.service.ts [hunk 2]
```diff
@@ -16,15 +22,24 @@ export class CreateGroupInvitationsService {
...
         map(successData),
-        map(objectToMap)
+        map(function (data: Object): Map<string, InvitationResult> {
+          return new Map<string, InvitationResult>(
+            Object.entries(data).map(
+              ([key, value]) => {
+                const result = InvitationResult[value as keyof typeof InvitationResult];
+                if (result == undefined)
+                  throw new Error(`Invitation of user ${key} returned an unexpected result`);
+                return [key, value];
```
Comment (smadbe) [inline src/app/modules/group/http-services/create-group-invitations.service.ts:40]: Ah I didn't know this weird syntax `InvitationResult[value as keyof typeof InvitationResult];`
Actually, I have not been clear but the service returns (following the API) `enum: [cycle, invalid, success, unchanged, not_found]` so you have to map these input values to the enum (cycle and invalid can be both mapped to error, the UI does not need to know more). So I expect more a switch-case which return the different values of the enum.

File: src/app/modules/group/http-services/create-group-invitations.service.ts [hunk 3]
```diff
@@ -16,15 +22,24 @@ export class CreateGroupInvitationsService {
...
       .pipe(
         map(successData),
-        map(objectToMap)
+        map(function (data: Object): Map<string, InvitationResult> {
+          return new Map<string, InvitationResult>(
+            Object.entries(data).map(
+              ([key, value]) => {
+                const result = InvitationResult[value as keyof typeof InvitationResult];
+                if (result == undefined)
+                  throw new Error(`Invitation of user ${key} returned an unexpected result`);
```
Comment (smadbe) [inline src/app/modules/group/http-services/create-group-invitations.service.ts:39]: in your error, you should print the value as well.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/group/http-services/create-group-invitations.service.ts (hunk 3 above)
Comment (GabrielVidal1) [reply src/app/modules/group/http-services/create-group-invitations.service.ts]: I used `JSON.stringify(value)` because the value is of type `any`

Comment (smadbe) [reply src/app/modules/group/http-services/create-group-invitations.service.ts]: Great 👍

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**